include/deal.II/lac/tensor_product_matrix.h

```diff
@@ -844,6 +854,80 @@ namespace internal
 
 
 
+    template <int n_rows_1d_templated, std::size_t dim, typename Number>
+    void
+    apply_inverse(Number *               dst,
```

Comment:
As this is an internal function, I think it would make sense to merge this new function with the other one, and simply add the arguments of both cases to the function arguments. You could then switch between the two by an `if` statement around the three nested `for` loops.